lib/tasks/update_resource_quantity.rake

```diff
@@ -0,0 +1,14 @@
+namespace :resources do
+  desc "Update resources with nil quantity"
+  task update_resource_quantity: :environment do
+    Resource.all.each do |r|
+      if r.quantity.nil? || r.used.nil?
+        if r.used.nil?
+          Resource.update(r.id, quantity: 0, used: 0)
```

Comment:
You can re-write this in a more compact and elegant way:

```
resource.used = 0 unless resource.used.present?
resource.quantity = resource.used unless resource.quantity present?
```

Then you can do `resource.save` with an if statement to make sure the resource is properly saved, and if not, inform the user about it at the end.